### ISSUE TITLE
Status reporting error - wrong server url

### ISSUE THREAD

**Issue by bbialas** [microsoft/cordova-plugin-code-push#567]
### Description
I use in my project pretty old (**~1.9.2-beta**) cordova-code-push version and since couple of days, updates are failing on reporting update status action.

```
[11:27:26] Checking for update.
[11:27:27] An error occurred while reporting status: {"status":0,"appVersion":"1.4","deploymentKey":"XXX"}
[11:27:27]  404: Cannot GET /v0.1/public/codepush/update_check?deployment_key=XXX&amp;app_version=1.4&amp;package_hash=YYY&amp;is_companion=false&amp;client_unique_id=ZZZ
[11:27:27] An error occurred during sync
```

After debugging and testing I realised, that the code-push is trying to fetch data from this url:
`https://codepush.azurewebsites.net/v0.1/public/codepush/update_check?deployment_key=XXX&amp;app_version=1.4&amp;package_hash=YYY&amp;is_companion=false&amp;client_unique_id=ZZZ`
Looks like this endpoint is not available... The response is **404 Not Found**

When I manually test latest server domain from the latest cordova-code-push version, it works:
`https://codepush.appcenter.ms/v0.1/public/codepush/update_check?deployment_key=XXX&amp;app_version=1.4&amp;package_hash=YYY&amp;is_companion=false&amp;client_unique_id=ZZZ`

**My questions are:**
1. Why `https://codepush.azurewebsites.net/` endpoint is down?
2. Could you maybe redirect it to the new `https://codepush.appcenter.ms/`?

It's a real **blocker** for older apps, because it's not possible to use code-push anymore, and I have thousands of users...

Please advise.

### Additional Information

* cordova-plugin-code-push version: **~1.9.2-beta**
* Cordova version: **9.0.0**
* Does this reproduce on a debug build or release build? **Release build**
* Does this reproduce on a simulator, or only on a physical device? **Tested on physical device**

**Comment by cjonsmith** [comment]
Hey so we've recently released a new version of the [code-push sdk](https://github.com/microsoft/code-push/tree/v3.0.1) that updated the endpoints to match the new Code Push service usage. At the same time, we released a new version of the [cordova plugin 1.12.0](https://github.com/microsoft/cordova-plugin-code-push/tree/v1.12.0) to update the Code Push server hostname to also use the new service to match the endpoint changes.

It seems that some older versions (anything before 1.9.6-beta) of the cordova plugin does [not use absolute versioning](https://github.com/microsoft/cordova-plugin-code-push/blob/v1.9.4-beta/plugin.xml#L9), so it's pulling the new version of the code-push package, but still using the old service's hostname, meaning it's using a mismatch of server hostname to endpoint names.

We're looking into redirecting these mismatched endpoints from the `https://codepush.azurewebsites.net` to our new `https://codepush.appcenter.ms` host, but in the meantime, if it's possible to release a new version of your app using version >= 1.9.6 of the cordova plugin, that should also resolve your issue.

**Comment by bbialas** [comment]
Thanks for the confirmation!
All clear, I already pushed new version of the app to Google/App Stores.

For binary updates I have a functionality in the app to display a popup to force users to open Google/App Store and update the app. The problem is, that this popup can be only activated via code-push update... 😞  Since the endpoint is wrong, code-push update is blocked and users are not able to download it...

Would be amazing if you could add this redirect... What do you think, when it could be fixed?
It's a life saving change 😄

**Comment by pfleidi** [comment]
@bbialas We're hoping to be able to deploy the change in the course of the day. We'll let you know when it is done.

Thanks for reporting this!

**Comment by cjonsmith** [comment]
Hey @bbialas, we just redeployed the CodePush server to proxy the new endpoints to the new service, so your issue should be resolved. We've validated that we could successfully download new updates using 1.9.2-beta of the cordova plugin. I'll go ahead and close this issue now, if you're still experiencing issues, feel free to let @pfleidi or I know here and we can re-open this issue. Apologies for the inconvenience!
